docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: docs
+title: Custom Browser Provider Plugin
+permalink: /documentation/extending-testcafe/custom-browser-provider-plugin/
+---
+# Custom Browser Provider Plugin
+
+TestCafe has built-in support for testing in a number of popular browsers (see [Using Browsers](../../using-testcafe/common-concepts/using-browsers.md)).
+If you need to use a different browser (for example, a browser from a cloud service), you can create a **custom browser provider plugin**.
+The provider should expose methods for performing common actions on the browser windows during testing: opening and closing it, taking screenshot, etc.
+
+To create a browser provider plugin, go through the following steps.
```

Comment:
Hi there. We've discussed how our plugin system will operate a couple of months ago, where have you been? And yes you should publish it, our plugin system is built on top of npm. If you don't wish to make it public you can use private npm repo or use `npm link`. But we don't encourage people to do so.